baggage/baggage.go

```diff
@@ -652,9 +652,65 @@ func validateValue(s string) bool {
 }
 
 func validateValueChar(c int32) bool {
-	return (c >= 0x23 && c <= 0x2b) ||
+	return c == 0x21 ||
+		(c >= 0x23 && c <= 0x2b) ||
 		(c >= 0x2d && c <= 0x3a) ||
 		(c >= 0x3c && c <= 0x5b) ||
-		(c >= 0x5d && c <= 0x7e) ||
-		c == 0x21
+		(c >= 0x5d && c <= 0x7e)
+}
+
+// valueEscape escapes the string so it can be safely placed inside a baggage value,
+// replacing special characters with %XX sequences as needed.
+//
+// The implementation is based on:
+// https://github.com/golang/go/blob/f6509cf5cdbb5787061b784973782933c47f1782/src/net/url/url.go#L285.
+func valueEscape(s string) string {
```

Comment:
I think we can export this function in order to have better resolution for https://github.com/open-telemetry/opentelemetry-go/issues/3685.

Of course the user would be able to call 

```go
baggage.NewMember(key, baggage.ValueEscape(val))
```

The benefit of it would be that the change would be backwards compatible.

I prefer to do this in a separate PR as I think it would be good to also update the documentation.

Reply:
FYI @yurishkuro

Reply:
No objections to this change, but I don't see how it would fix that other issue.

Reply:
It would not be a direct fix, but I think it would be a step towards resolving it.

E.g. I think that `baggage.ValueEscape` would have to be used here:

https://github.com/open-telemetry/opentelemetry-go/blob/885210bf33238f8c6cc94c3f3711b0036fbe77b1/bridge/opentracing/bridge.go#L78

Reply:
The main issue in #3685 is not about OT Bridge, but about a fundamental misunderstanding of the difference between Baggage API and W3C Baggage encoding rules. IMO it's an outright bug in this implementation. The Spec, unfortunately, was not prescriptive enough about that.